src/option.c

```diff
@@ -3864,6 +3864,16 @@ did_set_number_relativenumber(optset_T *args UNUSED)
     return NULL;
 }
 
+/*
+ * Process the updated 'msghistory' option value.
+ */
+    char *
+did_set_msghistory(optset_T *args UNUSED)
+{
+    check_msg_hist();
+    return NULL;
+}
```

Comment:
please move the error checking (max val/min val) here.

Reply:
Done.